src/Application/Application.csproj

```diff
@@ -7,7 +7,6 @@
 
     <ItemGroup>
         <ProjectReference Include="../Domain/Domain.csproj" />
-        <ProjectReference Include="../Infrastructure/Infrastructure.csproj" />
```

Comment:
The core way is correct. 
The description you made is also correct.

> In my mind the application depends on the infrastructure

What the infrastructure does is hooks up all the interfaces for the application. 
The application has no idea that that is going on. Interfaces just work.

This is why you have `IApplicationDbContext `for instance and not just `ApplicationDbContext`.

Something like this (50 hours in mspaint):

![image](https://user-images.githubusercontent.com/6649900/77345013-6fbd5100-6d0a-11ea-9a6f-5b37f081aad7.png)


A point of contention could be whether to put the .AddInfrastructure inside of the Application or the Startup.cs directly.

Reply:
> What the infrastructure does is hooks up all the interfaces for the application. The application has no idea that that is going on. Interfaces just work.

But currently we setup the `IServiceCollection` in all project separately, for example:

[src/Application/DependencyInjection.cs][1]

---

So we would have to move the interfaces in the following folder into the application project?

[src/Infrastructure/Common/Interfaces][2]

  [1]: https://github.com/codidact/core/blob/c5132e617e8c70ae3c72cb72f70c5f59a308fc89/src/Application/DependencyInjection.cs#L16-L20
  [2]: https://github.com/codidact/authentication/tree/80d95995b1e02f1bc6cc1a33333c1314575051fa/src/Infrastructure/Common/Interfaces

Reply:
> But currently we setup the IServiceCollection in all project separately, for example:

It doesn't matter. The DI register just knows whats up

> So we would have to move the interfaces in the following folder into the application project?

No, why did you assume that?

Reply:
> It doesn't matter. The DI register just knows whats up

I do not understand what you mean. I have seen other (closed source) repositories where all of the services are registered in the infrastructure project. That is similar to what you said before:

  > What the infrastructure does is hooks up all the interfaces for the application.

---

  > No, why did you assume that?

The application is currently not able to access `ISecretService` because that would be a circular dependency.

Reply:
> I do not understand what you mean. I have seen other (closed source) repositories where all of the services are registered in the infrastructure project. That is similar to what you said before:

So all the interfaces for the infrastructure are in the application project.
The hook up of the implementation of these services are in the infrastructure project.

So just to go through the flow:
1. We register the application logic in the application DependencyInjection.cs. These services use the interfaces from Common/Interfaces that is in the application project.
2. We register the infrastructure by taking all the interfaces in Common/Interfaces and connected them to the relevant implementations.

> The application is currently not able to access ISecretService because that would be a circular dependency.

ISecretService is in the Application project, not the infrastructure project.
Its in the Common/Interfaces folder.

Reply:
I think that clears everything up thank you!

  > ISecretService is in the Application project, not the infrastructure project.
Its in the Common/Interfaces folder.

Currently, they are in the infrastructure project, that is why I asked. I'll create a pull request.

Reply:
Oh yes, this is incorrect. I should've checked. Sorry about that